pkg/dao/uploads.go

```diff
@@ -0,0 +1,68 @@
+package dao
+
+import (
+	"context"
+
+	"github.com/content-services/content-sources-backend/pkg/models"
+	"github.com/content-services/content-sources-backend/pkg/pulp_client"
+	"gorm.io/gorm"
+)
+
+type uploadDaoImpl struct {
+	db         *gorm.DB
+	pulpClient pulp_client.PulpClient
+}
+
+func GetUploadDao(db *gorm.DB, pulpClient pulp_client.PulpClient) UploadDao {
+	return &uploadDaoImpl{
+		db:         db,
+		pulpClient: pulpClient,
+	}
+}
+
+func (t uploadDaoImpl) StoreFileUpload(ctx context.Context, orgID string, uploadUUID string, sha256 string, chunkSize int64) error {
+	var upload models.Upload
+
+	upload.OrgID = orgID
+	upload.UploadUUID = uploadUUID
+	upload.Sha256 = sha256
+	upload.ChunkSize = chunkSize
+
+	upload.ChunkList = []string{}
+
+	db := t.db.Model(models.Upload{}).WithContext(ctx).Create(&upload)
+	if db.Error != nil {
+		return db.Error
+	}
+
+	return nil
+}
+
+func (t uploadDaoImpl) GetExistingUploadIDAndCompletedChunks(ctx context.Context, orgID string, sha256 string, chunkSize int64) (string, []string, error) {
+	db := t.db.Model(models.Upload{}).WithContext(ctx)
+
+	var result models.Upload
+
+	db.Where("org_id = ?", orgID).Where("chunk_size = ?", chunkSize).Where("sha256 = ?", sha256).First(&result)
```

Comment:
i'd probably add an index on org_id, chunk_size, and sha256 to the db migration  to make this a bit more efficient

Reply:
Added, please take a look for a sanity check 😓 !